src/plugins/data/common/fields/types.ts

```diff
@@ -0,0 +1,44 @@
+/*
+ * Licensed to Elasticsearch B.V. under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch B.V. licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+export interface IFieldSubType {
+  multi?: { parent: string };
+  nested?: { path: string };
+}
+
+export interface IFieldType {
+  name: string;
+  type: string;
+  script?: string;
+  lang?: string;
+  count?: number;
+  // esTypes might be undefined on old index patterns that have not been refreshed since we added
+  // this prop. It is also undefined on scripted fields.
+  esTypes?: string[];
+  aggregatable?: boolean;
+  filterable?: boolean;
+  searchable?: boolean;
+  sortable?: boolean;
+  visualizable?: boolean;
+  readFromDocValues?: boolean;
+  scripted?: boolean;
+  subType?: IFieldSubType;
+  displayName?: string;
+  format?: any;
+}
```

Comment:
nit: Would it make more sense organizationally to keep these nested under `index_patterns` or elsewhere instead of a dedicated `fields` directory, since there isn't much here?

Not too concerned either way since it is just an implementation detail, only wanted to ask.

Reply:
Hm... looks like you are right, currently `fields` located inside `index_patterns`. fixed